### ISSUE TITLE
WiFi Connection Issues

### ISSUE THREAD

**Issue by ProjectMoon** [sailfish-on-dontbeevil/droid-config-pinephone#14]
I've encountered an issue where I'm unable to connect to WiFi networks anymore. I am not exactly sure what happened, but I was previously able to connect just fine, and then now I cannot connect to any. No connection seems to work, and all Sailfish says is that there was an error connecting. Forgetting the network and reconnecting does not work, neither does rebooting.

Presumably there's somewhere I can dig into the log files for internet connectivity. Where to start looking?

**Comment by ProjectMoon** [comment]
So I managed to fix this by turning of Connman's DNS proxy. I noticed that the phone had no DNS, By creating a file at `/var/lib/environment/connman/nodnsproxy.conf` with the contents:

```
CONNMAN_ARGS="--nodnsproxy"
```

The phone falls back to normal DHCP in `/etc/resolv.conf` and then DNS (and internet) work just fine.

**Comment by ProjectMoon** [comment]
More info on the inability to connect to wifi, from the system logs (journalctl):
```
wpa_supplicant[2664]: wlan0: Trying to associate with xx:xx:xx:xx:xx:xx (SSID='abc123') freq=2457 MHz)
wpa_supplicant[2664]: wlan0: Association request to the driver failed.
wpa_supplicant[2664]: wlan0: CTRL-EVENT-SSID-TEMP-DISABLED id=0 ssid="abc123" auth_failures=1 duration=10 reason=CONN_FAILED
```

**Comment by DylanVanAssche** [comment]
For some reason, I cannot reproduce this... Can you maybe explain a bit your network setup?

**Comment by ProjectMoon** [comment]
It is a mesh network, using Orbi routers. I suspect this issue is at the driver level, as I see similar results and the exact same logs in the syslog on Ubuntu Touch.

Are there any log levels to turn up in dmesg/syslog that could produce more useful output?

**Comment by DylanVanAssche** [comment]
The driver of the WiFi/BT chip is not great at the moment. A better driver is in the works I heard.
So I guess we should wait until the better driver comes into play (with power management!)

**Comment by piggz** [comment]
@ProjectMoon can you confirm this is still an issue with the latest release?

**Comment by ProjectMoon** [comment]
I tried Sailfish OS the other day, and the WiFi seems much more stable than it was in the beginning. Instead of dropping completely and never turning back on, it often gets very slow, sometimes disconnects (especially after screen off), and usually comes back. It did die once though.

But I suspect this is a driver issue for the PinePhone, less so than SFOS.

**Comment by ProjectMoon** [comment]
Current versions of the image have extremely unstable WiFI when I test them. More unstable than it used to be even. Is anyone else experiencing this?

**Comment by simonschmeisser** [comment]
Yes, same here. It often disconnects, sometimes I can "restart" it by toggling the button but sometimes then it doesn't want to re-connect at all

**Comment by DylanVanAssche** [comment]
WiFi is not very great on the dev phone I have (1.0).
But I think this happens on other OSes too? Can anyone please test if other OSes are suffering from the same issue?

**Comment by ProjectMoon** [comment]
All signs point to it being related to the wifi driver, yes. When I first opened this, it was January and Sailfish OS is the one I tested first.

There are specifically problems related to mesh networks.

But I do think distros like Ubuntu Touch have fairly reliable wifi, at least after a warm reboot. Is the SFOS port using the latest Pinephone wifi driver?

**Comment by piggz** [comment]
We are using the new driver iirc

**Comment by Qiangong2** [comment]
Still having the wifi issue. I can connect to open networks, but not WPA2 networks. I can connect to WPA2 networks on UBPorts
